iceoryx_hoofs/source/relocatable_pointer/base_relative_pointer.cpp

```diff
@@ -159,25 +169,28 @@ BaseRelativePointer::id_t BaseRelativePointer::searchId(ptr_t ptr) noexcept
     return getRepository().searchId(ptr);
 }
 
+// NOLINTJUSTIFICATION NewType is trivially copyable, hence pass by value is preferred
+// NOLINTNEXTLINE(performance-unnecessary-value-param)
 bool BaseRelativePointer::isValid(id_t id) noexcept
 {
-    return getRepository().isValid(id);
+    return getRepository().isValid(static_cast<id_underlying_t>(id));
 }
 
-PointerRepository<BaseRelativePointer::id_t, BaseRelativePointer::ptr_t>& BaseRelativePointer::getRepository() noexcept
+PointerRepository<BaseRelativePointer::id_underlying_t, BaseRelativePointer::ptr_t>&
+BaseRelativePointer::getRepository() noexcept
 {
-    static PointerRepository<id_t, ptr_t> repository;
+    static PointerRepository<id_underlying_t, ptr_t> repository;
```

Comment:
Just wondering if the `PointerRepository` should also use `id_t`?

Reply:
I personally would prefer to consider this in a larger refactoring. It is implementation detail.

The overall design is not ideal, but it works and I see no critical defects nor static code analysis warnings.

Reply:
Could be done in future. It wasn't done now for the sake of easier refactoring.